streams/src/test/java/org/apache/kafka/streams/StreamsBuilderTest.java

```diff
@@ -433,6 +433,21 @@ public void shouldUseSerdesDefinedInMaterializedToConsumeGlobalTable() {
         }
     }
 
+    @Test
+    public void shouldThrowOnVersionedStoreSupplierForGlobalTable() {
```

Comment:
Thanks for adding this. You'll also want to remove the existing integration test in VersionedKeyValueStoreIntegrationTest, and the relevant unit tests in KStreamGlobalKTableJoinTest and KStreamGlobalKTableLeftJoinTest.